Expand MiningStatistic component functionality: Introduce BoxDisplayer

shaavan opened the pull request:
This PR, along with #535

resolves #448

Introduce the BoxDisplay component to display the percentage for the user.

**Screenshot:**

|Green Background|Transparent Background|
|---|---|
|![Screenshot 2023-06-30 at 7 09 47 PM](https://github.com/saving-satoshi/saving-satoshi/assets/85434418/714dd35c-8612-497b-a33d-d93c22f90778)|![Screenshot 2023-06-30 at 7 10 00 PM](https://github.com/saving-satoshi/saving-satoshi/assets/85434418/54522ac3-7454-4ee8-a25e-3e57c5cf5dd5)|

benalleng commented on content/lessons/chapter-2/mining-1.tsx:
```diff
@@ -295,6 +296,17 @@ export default function Mining1({ lang }) {
               />
             </Card>
           </div>
+          <Card className="flex" highlight={nonceHighlight}>
+            <TitleCard title={'Hash Rate'} disabled={false}></TitleCard>
+            <BoxDisplayer className="ml-10">
+              <span>25.00%</span>
+            </BoxDisplayer>
+            <HashFrequency
+              disabled={nonce === 0}
+              step={step}
+              hashPower={hashPower}
+            />
+          </Card>
```
``` suggestion
```

shaavan replied:
Addressed in 8852c948c97a12026a90871b9b7c7208e8ebd52b
Thanks!

benalleng commented on content/lessons/chapter-2/mining-1.tsx:
```diff
@@ -11,6 +11,7 @@ import {
   TitleCard,
   HashFrequency,
   StartButton,
+  BoxDisplayer,
```
``` suggestion
```

shaavan replied:
Addressed in 8852c948c97a12026a90871b9b7c7208e8ebd52b
Thanks for reviewing!

benalleng reviewed (approved):
LGTM! lets just remove the test in mining-1 and then we should be good to merge!

shaavan commented:
Updated from 8eadabd2dc3380a6befabdef54228331f03209b5 to 8852c948c97a12026a90871b9b7c7208e8ebd52b ([pr540.01](https://github.com/shaavan/saving-satoshi/commits/pr540.01) -> [pr540.02](https://github.com/shaavan/saving-satoshi/commits/pr540.02), [diff](https://github.com/shaavan/saving-satoshi/compare/pr540.01..pr540.02))
Addressed @benalleng comment

- Removed test commit.
